TRLevelControl/Control/TR3/TR3MapControl.cs

```diff
@@ -0,0 +1,127 @@
+
+    protected override string ConvertKey(TR3Type key)
+        => _typeToKeyMap[key];
+
+    private static readonly Dictionary<string, TR3Type> _keyToTypeMap = new()
+    {
+        ["DUMMY"] = default,
+        ["AIRLOCK_SWITCH"] = TR3Type.ValveWheelOrPulley,
+        ["ANIMATING1"] = TR3Type.Animating1,
+        ["ANIMATING2"] = TR3Type.Animating2,
+        ["ANIMATING3"] = TR3Type.Animating3,
+        ["ANIMATING4"] = TR3Type.Animating4,
+        ["ANIMATING5"] = TR3Type.Animating5,
+        ["ANIMATING6"] = TR3Type.Animating6,
+        ["BIG_ROLLING_BALL"] = TR3Type.GiantBoulder,
+        ["BLADE"] = TR3Type.CircularBlade,
+        ["BRIDGE_FLAT"] = TR3Type.BridgeFlat,
+        ["BRIDGE_TILT1"] = TR3Type.BridgeTilt1,
+        ["BRIDGE_TILT2"] = TR3Type.BridgeTilt2,
+        ["CEILING_SPIKES"] = TR3Type.SpikyVertWallOrTunnelBorer,
+        ["COBRA"] = TR3Type.Cobra,
+        ["CROCODILE"] = TR3Type.Croc,
+        ["DIVER"] = TR3Type.ScubaSteve,
+        ["DOG"] = TR3Type.Dog,
+        ["DOOR_TYPE1"] = TR3Type.Door1,
+        ["DOOR_TYPE2"] = TR3Type.Door2,
+        ["DOOR_TYPE3"] = TR3Type.Door3,
+        ["DOOR_TYPE4"] = TR3Type.Door4,
+        ["DOOR_TYPE5"] = TR3Type.Door5,
+        ["DOOR_TYPE6"] = TR3Type.Door6,
+        ["DOOR_TYPE7"] = TR3Type.Door7,
+        ["DOOR_TYPE8"] = TR3Type.Door8,
+        ["EXTRAFX6"] = TR3Type.ElectricalSwitchBox,
+        ["FALLING_BLOCK"] = TR3Type.FallingBlock,
+        ["FALLING_CEILING1"] = TR3Type.FallingCeiling,
+        ["FAN"] = TR3Type.DamagingAnimating1,
+        ["FIREHEAD"] = TR3Type.FireBreathingDragonStatue,
+        ["ICICLES"] = TR3Type.DetachableStalactites,
+        ["ICON_PICKUP1_ITEM"] = TR3Type.Infada_P,
+        ["ICON_PICKUP1_OPTION"] = TR3Type.Infada_M_H,
+        ["KEY_HOLE1"] = TR3Type.Keyhole1,
+        ["KEY_HOLE2"] = TR3Type.Keyhole2,
+        ["KEY_HOLE3"] = TR3Type.Keyhole3,
+        ["KEY_HOLE4"] = TR3Type.Keyhole4,
+        ["KEY_ITEM1"] = TR3Type.Key1_P,
+        ["KEY_ITEM2"] = TR3Type.Key2_P,
+        ["KEY_ITEM3"] = TR3Type.Key3_P,
+        ["KEY_ITEM4"] = TR3Type.Key4_P,
+        ["KEY_OPTION1"] = TR3Type.Key1_M_H,
+        ["KEY_OPTION2"] = TR3Type.Key2_M_H,
+        ["KEY_OPTION3"] = TR3Type.Key3_M_H,
+        ["KEY_OPTION4"] = TR3Type.Key4_M_H,
+        ["LIZARD_MAN"] = TR3Type.LizardMan,
+        ["LON_MERCENARY1"] = TR3Type.LondonMerc,
+        ["MONKEY"] = TR3Type.Monkey,
+        ["MOVABLE_BLOCK"] = TR3Type.PushableBlock1,
+        ["MOVABLE_BLOCK2"] = TR3Type.PushableBlock2,
+        ["MP1"] = TR3Type.MPWithStick,
+        ["PASSPORT_OPTION"] = TR3Type.PassportOpening_H,
+        ["PENDULUM"] = TR3Type.SwingingThing,
+        ["PICKUP_ITEM1"] = TR3Type.Quest1_P,
+        ["PICKUP_OPTION1"] = TR3Type.Quest1_M_H,
+        ["PLAYER_1"] = TR3Type.CutsceneActor1,
+        ["PLAYER_2"] = TR3Type.CutsceneActor2,
+        ["PLAYER_3"] = TR3Type.CutsceneActor3,
+        ["PLAYER4"] = TR3Type.CutsceneActor4,
+        ["PLAYER_4"] = TR3Type.CutsceneActor_4,
```

Comment:
Why does this have `_`?

Reply:
There seems to be a duplicate of sorts in the map files for TR3, with `PLAYER4` and `PLAYER_4`. Because we need unique keys for the flipped version of the dictionary, I just added the new placeholder type. It's not something we'll be changing to be honest, it's mainly just to prove we can match everything up for the tests.